File: php/rooms/options/RoomWithCreature.php
<?php

namespace rooms\options;

use logic\GameMaster;
use monsters\options\Creature;
use monsters\Hero;

class RoomWithCreature extends RoomDecorator {
    private $creature;

    public function __construct(GameMasterable $explorable, Creature $creature) {
        parent::__construct($explorable);
        $this->creature = $creature;
    }

    public function explore(Hero $hero) {
        $this->room->getMuster()->performBattle($hero, $this->creature);
        $this->room->explore($hero);
    }

    public function getNextRooms() {
        return $this->room->getNextRooms();
    }

    public function getRoomType() {
        return $this->room->getRoomType();
    }

    public function getMuster() {
        return $this->room->getMuster();
    }
}
?>

File: java/rooms/options/RoomWithCreature.java
package rooms.options;

import logic.GameMaster;
import monsters.options.Creature;
import monsters.Hero;

public class RoomWithCreature extends RoomDecorator{
    private Creature creature;

    public RoomWithCreature(GameMasterable explorable, Creature creature) {
        super(explorable);
        this.creature = creature;
    }


    @Override
    public void explore(Hero hero) {
        room.getMuster().performBattle(hero,creature);
        room.explore(hero);
    }

    @Override
    public GameMasterable[] getNextRooms() {
        return room.getNextRooms();
    }

    @Override
    public RoomType getRoomType() {
        return room.getRoomType();
    }

    @Override
    public GameMaster getMuster() {
        return room.getMuster();
    }

}
